### ISSUE TITLE
Invalid escape sequences in s"" interpolated strings throw at runtime, inconsistent with f"" and double quoted strings

### ISSUE THREAD

**Issue by scabug** [scala/bug#9234]
Compiling 

```scala
def s = "bad escape: \z"
```

produces a compile-time error like:
```scala
error: invalid escape character
       def s = "bad escape: \z"
                             ^
```

The standard s"" and f"" string interpolators use the same escape sequences as normal double quoted strings.  Like double quoted strings the f"" interpolator reports bad sequences at compile time.

Inconsistently, the s"" string interpolator reports bad escape sequences at runtime by throwing an exception. 

For example, compiling the following code succeeds:

```scala
def f = s"bad escape: \z"
```

At run time invoking f will produce an exception:
```scala
scala.StringContext$InvalidEscapeException: invalid escape '\z' not one of [\b, \t, \n, \f, \r, \\, \", \'] at index 12 in "bad escape: \z". Use \\ for literal \.
  at scala.StringContext$.loop$1(StringContext.scala:234)
  at scala.StringContext$.replace$1(StringContext.scala:244)
  at scala.StringContext$.treatEscapes0(StringContext.scala:248)
  at scala.StringContext$.treatEscapes(StringContext.scala:193)
  at scala.StringContext$$anonfun$s$1.apply(StringContext.scala:94)
  at scala.StringContext$$anonfun$s$1.apply(StringContext.scala:94)
  at scala.StringContext.standardInterpolator(StringContext.scala:122)
  at scala.StringContext.s(StringContext.scala:94)
  at .f(<console>:7)
```

I would prefer that the s"" interpolator report bad escape errors as early as possible, consistent with standard double quoted strings and f"" interpolated strings.

**Comment by latkin** [comment]
Just got hit by this, would love to see a fix.

**Comment by NthPortal** [comment]
It looks like this may be fixed in 2.13 (see scala/scala-dev#475), but can't be in 2.12 for bincompat reasons

**Comment by som-snytt** [comment]
There is a trivial test test/files/neg/t6631.scala
```
scala 2.13.2> val x = 42
val x: Int = 42

scala 2.13.2> s"$x"
val res0: String = 42

scala 2.13.2> s"$x\t$x"
val res1: String = 42   42

scala 2.13.2> s"$x\x$x"
                ^
              error: invalid escape '\x' not one of [\b, \t, \n, \f, \r, \\, \", \', \uxxxx] at index 0 in "\x". Use \\ for literal \.

scala 2.13.2> def f = () => s"$x\x$x"
                              ^
              error: invalid escape '\x' not one of [\b, \t, \n, \f, \r, \\, \", \', \uxxxx] at index 0 in "\x". Use \\ for literal \.
```
https://github.com/scala/scala/pull/7678
